src/main/java/net/minecraftforge/common/world/biomes/BiomeModifications.java

```diff
@@ -0,0 +1,188 @@
+    {
+        this.temperature = temperature;
+        return this;
+    }
+
+    public BiomeModifications modifyDownfall(UnaryOperator<Float> downfall)
+    {
+        this.downfall = downfall;
+        return this;
+    }
+
+    public BiomeModifications modifyRainType(UnaryOperator<Biome.RainType> precipitation)
+    {
+        this.precipitation = precipitation;
+        return this;
+    }
+
+    public BiomeModifications modifyTemperatureModifier(UnaryOperator<Biome.TemperatureModifier> temperatureModifier)
+    {
+        this.temperatureModifier = temperatureModifier;
+        return this;
+    }
+
+    public BiomeModifications modifyCategory(UnaryOperator<Biome.Category> category)
+    {
+        this.category = category;
+        return this;
+    }
+
+    public BiomeModifications modifyDepth(UnaryOperator<Float> depth)
+    {
+        this.depth = depth;
+        return this;
+    }
+
+    public BiomeModifications modifyScale(UnaryOperator<Float> scale)
+    {
+        this.scale = scale;
+        return this;
+    }
+
+    public BiomeModifications modifyBiomeAmbience(UnaryOperator<BiomeAmbience> effects)
+    {
+        this.effects = effects;
+        return this;
+    }
+
+    public BiomeModifications modifySurfaceBuilder(UnaryOperator<Supplier<ConfiguredSurfaceBuilder<?>>> surfaceBuilder)
+    {
+        this.surfaceBuilder = surfaceBuilder;
+        return this;
+    }
+
+    public BiomeModifications modifyCarvers(UnaryOperator<Map<GenerationStage.Carving, List<Supplier<ConfiguredCarver<?>>>>> carvers)
+    {
+        this.carvers = carvers;
+        return this;
+    }
+
+    public BiomeModifications modifyFeatures(UnaryOperator<List<List<Supplier<ConfiguredFeature<?, ?>>>>> features)
+    {
+        this.features = features;
+        return this;
+    }
+
+    public BiomeModifications modifyStructures(UnaryOperator<List<Supplier<StructureFeature<?, ?>>>> structures)
+    {
+        this.structures = structures;
+        return this;
+    }
+
+    public BiomeModifications modifyCreatureSpawnProbability(UnaryOperator<Float> creatureSpawnProbability)
+    {
+        this.creatureSpawnProbability = creatureSpawnProbability;
+        return this;
+    }
+
+    public BiomeModifications modifySpawners(UnaryOperator<Map<EntityClassification, List<MobSpawnInfo.Spawners>>> spawners)
+    {
+        this.spawners = spawners;
+        return this;
+    }
+
+    public BiomeModifications modifySpawnCosts(UnaryOperator<Map<EntityType<?>, MobSpawnInfo.SpawnCosts>> spawnCosts)
+    {
+        this.spawnCosts = spawnCosts;
+        return this;
+    }
+
+    public BiomeModifications modifyPlayerSpawnFriendly(UnaryOperator<Boolean> playerSpawnFriendly)
+    {
+        this.playerSpawnFriendly = playerSpawnFriendly;
+        return this;
+    }
+
+    static Biome performModifications(List<BiomeModifications> modifications, Biome base)
+    {
+        return modifications.isEmpty() ? base :
+                new Biome.Builder()
```

Comment:
I feel like this method would be more readable if you just split out the `? :` into an if statement